pkg/inventory/model/client.go

```diff
@@ -208,24 +207,28 @@ func (r *Client) Insert(model Model) error {
 func (r *Client) Update(model Model) error {
 	r.Lock()
 	defer r.Unlock()
-	model.SetPk()
 	table := Table{}
 	if r.tx == nil {
-		r.mutex.Lock()
-		defer r.mutex.Unlock()
+		r.dbMutex.Lock()
+		defer r.dbMutex.Unlock()
 		table.DB = r.db
 	} else {
 		table.DB = r.tx
 	}
-	err := table.Update(model)
+	current := r.journal.copy(model)
```

Comment:
Grabbing the _current_ version of the model before updated just to pass to the journal.